napari/utils/progress.py

```diff
@@ -0,0 +1,130 @@
+import inspect
+from typing import Iterable, Optional
+
+from PyQt5 import QtCore
+from PyQt5.QtWidgets import (
+    QApplication,
+    QHBoxLayout,
+    QLabel,
+    QProgressBar,
+    QWidget,
+)
+
+from .._qt.utils import get_viewer_instance
+
+
+def get_pbar():
+    pbar = ProgressBar()
+    viewer_instance = get_viewer_instance()
+    viewer_instance.activityDock.widget().layout.addWidget(pbar.baseWidget)
+
+    return pbar
+
+
+def get_calling_function_name(max_depth: int):
+    """Inspect stack up to max_depth and return first function name outside of progress.py"""
+    for finfo in inspect.stack()[2:max_depth]:
+        if not finfo.filename.endswith("progress.py"):
+            return finfo.function
+
+    return None
+
+
+class progress:
+    def __init__(
+        self,
+        iterable: Optional[Iterable] = None,
+        total: Optional[int] = None,
+        description: Optional[str] = None,
```

Comment:
```suggestion
        desc: Optional[str] = None,
        total: Optional[int] = None,
```

I think it's very possible that we may someday want `progress` to be a subclass of `tqdm` (as I did in magicgui, to benefit from all the additional niceties that they offer when wrapping an iterable (like min/max display intervals), not to mention the fact that we could just then fallback to ipython progressbar if no viewer exists.  So I'd propose that we stick to the naming and order of [their `__init__` kwargs](https://github.com/tqdm/tqdm/blob/4591083884c8ca079ccd1192c079113c7baeadc8/tqdm/std.py#L842-L848) and attribute names as much as possible, just in case.